spark/spark-spa/src/test/groovy/com/thoughtworks/go/spark/spa/ElasticAgentsControllerTest.groovy

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.spark.spa
+
+
+import com.thoughtworks.go.spark.ControllerTrait
+import com.thoughtworks.go.spark.GroupAdminUserSecurity
+import com.thoughtworks.go.spark.SecurityServiceTrait
+import com.thoughtworks.go.spark.spring.SPAAuthenticationHelper
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+
+import static org.mockito.MockitoAnnotations.initMocks
+
+class ElasticAgentsControllerTest implements ControllerTrait<ElasticAgentsController>, SecurityServiceTrait {
+  @BeforeEach
+  void setUp() {
+    initMocks(this)
+  }
+
+  @Override
+  ElasticAgentsController createControllerInstance() {
+    return new ElasticAgentsController(new SPAAuthenticationHelper(securityService, goConfigService), templateEngine)
+  }
+
+  @Nested
+  class Index {
+    @Nested
+    class Security implements SecurityTestTrait, GroupAdminUserSecurity {
```

Comment:
@akshaydewan -- I am a bit confused with this page's authorization.
So,
- The cluster profiles can only be accessed by super admins
- The elastic agent profiles can be accessed by pipeline group admins

On page load, we make an API call to fetch cluster profiles and elastic agent profiles. As a pipeline group admin user will not have access to cluster profiles API, the API would fail with a 403 error resulting in page load failure.

I haven't tested this behavior with these changesets, but I can confirm this behavior on the existing SPA.

Reply:
You're right, the page won't work unless a user is a super admin. I can't think of a good answer until we implement Policies for Cluster Profiles and Elastic Profiles. Should we keep it the way it is right now until the policies are implemented?

cc: @maheshp @arvindsv @adityasood

Reply:
Currently a Pipeline Group admin gets an error on visiting the Elastic Agent Configuration page. I would prefer showing this link only to System Admins for now. 

Once we support granular auth for these configurations we can decide on the appropriate behavior, that will most likely be showing information based on permissions.